libs/experimental/langchain_experimental/chain_of_verification/base.py

```diff
@@ -79,14 +83,20 @@ def _call(
         self,
         inputs: Dict[str, Any],
         run_manager: Optional[CallbackManagerForChainRun] = None,
-    ) -> Dict[str, str]:
-        _run_manager = run_manager or CallbackManagerForChainRun.get_noop_manager()
+    ) -> Dict[str, Any]:
+        if run_manager:
+            run_manager.on_text("Running Chain of Verification...\n")
         query = inputs[self.input_key]
+        output = self.cove_chain.invoke(query)
 
-        output = self.cove_chain.invoke(
-            {"query": query}, callbacks=_run_manager.get_child()
-        )
-        return {self.output_key: output["result"]}
+        # Log intermediate steps
+        # TODO: find a way to log during the execution of the chain
+        if run_manager:
+            run_manager.on_text(f"Baseline response: {output['baseline_response']}\n\n")
```

Comment:
I think this would work fine for logging :)
A very minor detail, maybe add a newline after the Baseline response? `f"Baseline response: \n{output['baseline_response']}\n\n"`

The below would log:

```python
from langchain.callbacks import StdOutCallbackHandler
handler = StdOutCallbackHandler()
cove_chain = CoVeChain.from_llm(llm=llm, callbacks=[handler])
cove_chain.invoke("List 5 politicians born in London")
```

```
> Entering new CoVeChain chain...
Running Chain of Verification...
Baseline response: 1. Boris Johnson
2. Sadiq Khan
3. David Cameron
4. Theresa May
5. Jeremy Corbyn

Verification questions/answers: 
Question: Who is a politician born in London?
Answer: One example of a politician born in London is Boris Johnson, who was born in New York City but moved to London at a young age and has served as the Mayor of London and Prime Minister of the United Kingdom.
```